### PR TITLE
Remove duplicate macros between j9nonbuilder.h and j9consts.h

### PR DESCRIPTION
**Pull request by AidanHa**
-There were various duplicate macros in j9consts.h and j9nonbuilder.h. Since j9nonbuilder.h seemed to be more up to date, I removed several constants from j9consts.h, including the following:

Note: This is the second pull request attempt.

J9_JAVA_PUBLIC
J9_JAVA_PRIVATE
J9_JAVA_PROTECTED 0x4
J9_JAVA_STATIC (CARE IN STACK WALKER.JAVA)
J9_JAVA_FINAL
J9_JAVA_CLASS_RAM_SHAPE_SHIFT
J9_JAVA_CLASS_REFERENCE_SHIFT
J9_JAVA_SYNC
J9_JAVA_NATIVE
J9_JAVA_GETTER_METHOD
J9_JAVA_INTERFACE
J9_JAVA_UNUSED_0X200
J9_JAVA_ABSTRACT
J9_JAVA_FORWARDER_METHOD
J9_JAVA_EMPTY_METHOD
J9_JAVA_CLASS_DEPTH_MASK
J9_JAVA_CLASS_ARRAY
J9_JAVA_CLASS_RAM_ARRAY
J9_JAVA_METHOD_VTABLE
J9_JAVA_METHOD_HAS_EXCEPTIONS
J9_JAVA_CLASS_PRIMITIVE_TYPE
J9_JAVA_CLASS_UNSAFE
J9_JAVA_METHOD_FRAME_ITERATOR_SKIP
J9_JAVA_METHOD_CALLER_SENSITIVE
J9_JAVA_CLASS_BYTECODES_MODIFIED
J9_JAVA_CLASS_OWNABLE_SYNCHRONIZER
J9_JAVA_CLASS_HAS_EMPTY_FINALIZE
J9_JAVA_CLASS_HAS_JDBC_NATIVES
J9_JAVA_METHOD_OBJECT_CONSTRUCTOR
J9_JAVA_CLASS_GC_SPECIAL
J9_JAVA_CLASS_HAS_VERIFY_DATA
J9_JAVA_METHOD_UNUSED_1000000
J9_JSVS_CLASS_CONTENDED
J9_JAVA_CLASS_HAS_FINAL_FIELDS
J9_JAVA_METHOD_HAS_GENERIC_SIGNATURE
J9_JAVA_CLASS_HOT_SWAPPED_OUT
J9_JAVA_CLASS_HAS_CLINIT
J9_JAVA_CLASS_DYING
J9_JAVA_CLASS_REFERENCE_WEAK
J9_JAVA_FIELD_UNUSED_10000000
J9_JAVA_CLASS_REFERENCE_SOFT
J9_JAVA_CLASS_REFERENCE_MASK
J9_JAVA_CLASS_REFERENCE_PHANTOM
J9_JAVA_CLASS_CLONEABLE
J9_JAVA_CLASS_ROMRAMMASK

-I renamed the constants in various files. I did not touch anything in any of the DDR files @tajila
-There were alot of constants to renamed, so hopefully I did not made any typos.
I also added <#include "j9nonbuilder.h"> where necessary.

I also updated the Copyright comments where necessary, and modified the .gitignore file to ignore any .vscode folders

### CODE REVIEW

**Comment by AidanHa** [comment]
Ah, thank you for the link @fjeremic I will edit the commit messages.

The multiple unnecessary sign offs come from me squashing multiple smaller commits together. Will I only need to sign off one commit in this case (sign off the latest commit)?

**Comment by fjeremic** [comment]
> Will I only need to sign off one commit in this case (sign off the latest commit)?

That's correct.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2000, 2018 IBM Corp. and others
+# Copyright (c) 2000, 2019 IBM Corp. and others
 #
```
Comment (tajila) [inline .gitignore]: can you please revert the changes in this file

File: runtime/compiler/env/VMJ9.cpp [hunk 2]
```diff
@@ -32,6 +32,7 @@
 #include "cache.h"
 #include "fastJNI.h"
 #include "j9cfg.h"
+
```
Comment (tajila) [inline runtime/compiler/env/VMJ9.cpp]: please remove this added whitespace

File: runtime/compiler/il/symbol/J9Symbol.cpp [hunk 3]
```diff
@@ -35,6 +35,7 @@
 #include <stddef.h>                            // for size_t
 #include <stdint.h>                            // for uint32_t, int32_t, etc
 #include <string.h>                            // for memcmp, strncmp
+
```
Comment (tajila) [inline runtime/compiler/il/symbol/J9Symbol.cpp]: same here

File: runtime/compiler/runtime/SymbolValidationManager.hpp [hunk 4]
```diff
@@ -32,6 +32,7 @@
 #include "env/jittypes.h"                  // for uintptrj_t
 #include "j9.h"
 #include "j9nonbuilder.h"
+#include "j9consts.h"
```
Comment (tajila) [inline runtime/compiler/runtime/SymbolValidationManager.hpp]: was this needed in this here

File: runtime/gc_base/ClassLoaderManager.hpp [hunk 5]
```diff
@@ -32,7 +32,6 @@
 #include "j9.h"
 #include "j9cfg.h"
 
-
```
Comment (tajila) [inline runtime/gc_base/ClassLoaderManager.hpp]: undo this change

File: runtime/gc_glue_java/ConcurrentMarkingDelegate.hpp [hunk 6]
```diff
@@ -33,6 +33,7 @@
 #include "j9modron.h"
 #include "j9nongenerated.h"
 #include "j9nonbuilder.h"
+#include "j9consts.h"
```
Comment (tajila) [inline runtime/gc_glue_java/ConcurrentMarkingDelegate.hpp]: what was this needed for?

File: runtime/gc_glue_java/MarkingDelegate.hpp [hunk 7]
```diff
@@ -25,6 +25,7 @@
 #define MARKINGDELEGATE_HPP_
 
 #include "j9nonbuilder.h"
+#include "j9consts.h"
```
Comment (tajila) [inline runtime/gc_glue_java/MarkingDelegate.hpp]: same here

File: runtime/gc_glue_java/ObjectModelDelegate.hpp [hunk 8]
```diff
@@ -24,6 +24,7 @@
 #define OBJECTMODELDELEGATE_HPP_
 
 #include "j9nonbuilder.h"
+#include "j9consts.h"
```
Comment (tajila) [inline runtime/gc_glue_java/ObjectModelDelegate.hpp]: ^^

File: runtime/oti/j9nonbuilder.h [hunk 9]
```diff
@@ -33,6 +33,7 @@
 #include "j9vmconstantpool.h"
 #include "j9consts.h"
 #include "jvmti.h"
+#include "omrsrp.h"
```
Comment (tajila) [inline runtime/oti/j9nonbuilder.h]: is this still needed here, now that the macros are moved over to j9const.h ?

File: runtime/vm/guardedstorage.c [hunk 10]
```diff
@@ -24,6 +24,7 @@
 #include "j9gs.h"
 #include "j9port_generated.h"
 #include "j9nonbuilder.h"
+#include "j9consts.h"
```
Comment (tajila) [inline runtime/vm/guardedstorage.c]: is this needed?

**Comment by AidanHa** [comment]
@tajila for the some of the #include "j9consts.h" calls, I just wanted to make sure that all files that included nonbuilder.h also had j9consts (kind of a surefire way to make sure there were no errors). I will now remove all the unessesary #includes.

File: sourcetools/j9constantpool/com/ibm/oti/VMCPTool/ConstantPoolStream.java [hunk 11]
```diff
@@ -388,4 +388,4 @@ public void comment(String string) {
 			out.print("\t/* " + string + " */");
 		}
 	}
-}
+}
```
Comment (tajila) [inline sourcetools/j9constantpool/com/ibm/oti/VMCPTool/ConstantPoolStream.java]: neeed to keep the newline

**Comment by tajila** [comment]
One minor comment rest LGTM

@dmitripivkine Please review the GC changes

@fjeremic Please review the JIT changes

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java [hunk 12]
```diff
@@ -47,7 +48,7 @@ protected GCClassArrayClassSlotIterator(J9ClassPointer clazz) throws CorruptData
 			slots.add(slot);
 			addresses.add(VoidPointer.cast(clazz.arrayClassEA()));
 		}
-		if(clazz.romClass().modifiers().allBitsIn(J9_JAVA_CLASS_ARRAY)) {
+		if(clazz.romClass().modifiers().allBitsIn(J9AccClassArray)) {
```
Comment (keithc-ca) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java]: This will break the ability to look at core files generated before this change.

**Comment by keithc-ca** [comment]
> > Will I only need to sign off one commit in this case (sign off the latest commit)?
> 
> That's correct.

No, each commit needs a `Signed-off-by:` line.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java (hunk 12 above)
Comment (AidanHa) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java]: Should I be reverting the macro name change here? Or is there a better fix to this problem?

Comment (keithc-ca) [reply debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java]: I spoke with @tajila who assures me that the values in the two classes are equal and so this should be compatible with older core files; I'll resume my review...

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java [hunk 13]
```diff
@@ -30,6 +30,7 @@
 import com.ibm.j9ddr.vm29.pointer.generated.J9ClassPointer;
 
 import static com.ibm.j9ddr.vm29.structure.J9Consts.*;
```
Comment (keithc-ca) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java]: I believe this import becomes unused: please remove it and similar imports elsewhere.

File: debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java [hunk 14]
```diff
@@ -30,6 +30,7 @@
...
+import static com.ibm.j9ddr.vm29.structure.J9JavaAccessFlags.*;
```
Comment (keithc-ca) [inline debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/j9/gc/GCClassArrayClassSlotIterator.java]: This file only uses one field of that class; it would be clearer to say:
```
import static com.ibm.j9ddr.vm29.structure.J9JavaAccessFlags.J9AccClassArray;
```

**Comment by AidanHa** [comment]
Sure, let me add the fixes and do a rebase to see if anything will change

**Comment by AidanHa** [comment]
I've rebased this branch to Eclipse:master just over an hour ago, and ran a build on my local machine (which ended up being successful) @keithc-ca. I've made changes to address the comments above.

**Comment by keithc-ca** [comment]
Jenkins test sanity zlinux,win jdk8,jdk11

**Comment by gacholio** [comment]
Reverted this due to DDR compile failures in the internal builds.

**Comment by keithc-ca** [comment]
@AidanHa This might work in reverse: removing uses of `J9JavaAccessFlags` in favour of using constants in `J9Consts`.

**Comment by AidanHa** [comment]
I'm not too familiar with the DDR code but from my point of view, I noticed there are no macros in `j9consts.h` that map to the ddr namespace (in other words, no there’s no code like
`/* @DDR_namespace: map_to_type=J9JavaAccessFlags */` in `j9consts.h`). Perhaps when the DDR is being compiled, then it doesn't even look at `j9consts.h`? If that is the case, where can I go to fix this?

**Comment by keithc-ca** [comment]
If there were no internal IBM builds, your change would have been fine. If we're going to consider eliminating duplicate constants, we must use only those that also *were* available in the past so that old core files produced by IBM builds can be analyzed.

**Comment by gacholio** [comment]
While I agree that's good practice, there will be circumstances where the back compatibility pain will be too high. You can always use the DDR that shipped with the drop that produced the core to analyze it.

**Comment by AidanHa** [comment]
image
We were able to use the new solution, and it successfully produced a class file S|J9JavaAccessFlags|J9JavaAccessFlagsPointer| for vm 28.  @keithc-ca, what should the next step be? 

The screenshot above is from the superset file, and it contains all the macros that we moved from j9nonbuilder.h

**Comment by AidanHa** [comment]
For context, I made a new file `j9javaaccessflags.h`, moved the macros into that file, and changed the `vmddrstructs.properties` file to build the macros from the new file.
